cmd/tsdb/main.go

```diff
@@ -31,19 +31,37 @@ import (
 	"time"
 
 	"github.com/go-kit/kit/log"
+	"github.com/m3db/m3/src/cluster/shard"
+	"github.com/m3db/m3/src/dbnode/persist/fs"
+	"github.com/m3db/m3/src/dbnode/persist/fs/commitlog"
+	"github.com/m3db/m3/src/dbnode/retention"
+	"github.com/m3db/m3/src/dbnode/sharding"
+	"github.com/m3db/m3/src/dbnode/storage"
+	"github.com/m3db/m3/src/dbnode/storage/namespace"
+	"github.com/m3db/m3/src/dbnode/storage/series"
+	m3dbTs "github.com/m3db/m3/src/dbnode/ts"
+	"github.com/m3db/m3x/ident"
+	xtime "github.com/m3db/m3x/time"
 	"github.com/pkg/errors"
 	"github.com/prometheus/tsdb"
 	"github.com/prometheus/tsdb/labels"
 	"gopkg.in/alecthomas/kingpin.v2"
 )
 
+var (
+	defaultM3DBNamespace = ident.StringID("defaultM3DBNamespace")
+	defaultSeriesID      = ident.StringID("default_m3db_id")
+	defaultNumShardsM3DB = 100
```

Comment:
Does m3db guarantee atomicity across shards? Scrapes are supposed to be atomic, which is why prometheus/tsdb only has a single shard.

To be fair, the actual MVCC isn't implemented yet (what's the status there @gouthamve?). But the resulting slowdown due to locking is already factored in. Would be interesting to get numbers with this set to `1` unless there's atomicity across shards already (or possible in the future).

Reply:
We don't have any atomicity guarantees (we don't have any concept of transactions within a shard or across shards). If a batch write fails (lets say half the write succeed and half fail for some weird reason), then you can retry the entire thing and if everything succeeds you'll be in the same state as if everything succeeded the first time, but if you don't retry then half of the datapoints will still be present.

I assume thats what you meant by atomic?

We could explore supporting that functionality, but we've just never had a reason for it before. Some data has always been better than no data for us.